File: dataingest-storm/src/main/java/com/xavient/dataingest/storm/spout/kafka/KafkaSpoutFactory.java
package com.xavient.dataingest.storm.spout.kafka;

import com.xavient.dataingest.storm.constants.Constants;
import com.xavient.dataingest.storm.vo.AppArgs;

import storm.kafka.KafkaSpout;

public class KafkaSpoutFactory {

	public static KafkaSpout getKafkaSpout(AppArgs appArgs) {
    return KafkaInputSpout.getKafkaSpout(appArgs.getProperty(Constants.KAFKA_TOPIC),
        appArgs.getProperty(Constants.ZK_HOST), appArgs.getProperty(Constants.ZK_PORT, "2181"),
        appArgs.isRewind());
  }
}

File: dataingest-storm-druid/src/main/java/com/xavient/dataingest/druid/Spouts/KafkaSpoutFactory.scala
package com.xavient.dataingest.druid.Spouts

import backtype.storm.spout.SchemeAsMultiScheme;
import storm.kafka.BrokerHosts;
import storm.kafka.KafkaSpout;
import storm.kafka.SpoutConfig;
import storm.kafka.StringScheme;
import storm.kafka.ZkHosts;
import com.xavient.dataingest.druid.Utils.Constants
import com.xavient.dataingest.druid.Utils.Property

object KafkaSpoutFactory {

  def getKafkaSpout(): KafkaSpout = {
    return KafkaInputSpout.getKafkaSpout(Property.getProperty(Constants.KAFKA_TOPIC),
      Property.getProperty(Constants.KAFKA_ZOOKEEPER), Property.getProperty(Constants.KAFKA_PORT), false);

  }

}